src/main/java/training/supportbank/DataHandler.java

```diff
@@ -0,0 +1,77 @@
+package training.supportbank;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.*;
+
+public class DataHandler {
+    public static List<Map<String, String>> dataConverter(BufferedReader reader) throws IOException {
+        String transactionLine;
+        List<Map<String, String>> transactionArray = new ArrayList<>();
+
+        while ((transactionLine = reader.readLine()) != null) {
+            Map<String, String> transactionMap = new LinkedHashMap<>();
+            String[] transaction = transactionLine.split(",");
+            transactionMap.put("Date", transaction[0].replace("\"", ""));
+            transactionMap.put("From", transaction[1]);
+            transactionMap.put("To", transaction[2]);
+            transactionMap.put("Narrative", transaction[3]);
+            transactionMap.put("Amount", transaction[4].replace("\"", ""));
+            transactionArray.add(transactionMap);
+        }
+        return transactionArray;
+    }
+
+    public static Set<String> extractNames(List<Map<String, String>> inputArray) {
+        String fromName, toName;
+        Set<String> setOfNames = new HashSet<>();
+
+        for (int i = 0; i < inputArray.size(); i++) {
+            fromName = inputArray.get(i).get("From");
+            toName = inputArray.get(i).get("To");
+            setOfNames.add(fromName);
+            setOfNames.add(toName);
+        }
+        return setOfNames;
+    }
+
+    public static Map<String, BigDecimal> calculateBalance(List<Map<String, String>> inputArray, Set<String> setOfNames) {
+        Map<String, BigDecimal> finalOutput = new LinkedHashMap<>();
+        finalOutput = getValues(inputArray, setOfNames, finalOutput,"From");
+        finalOutput = getValues(inputArray, setOfNames, finalOutput,"To");
+        return finalOutput;
+    }
+
+    private static Map<String, BigDecimal> getValues(List<Map<String, String>> inputArray, Set<String> setOfNames, Map<String, BigDecimal> outputMap, String mapKey) {
+        BigDecimal currVal, newVal, totalVal;
+
+        for (String name : setOfNames) {
+            for (int i = 0; i < inputArray.size(); i++) {
```

Comment:
We should be able to simplify this logic, but let's discuss it after we've simplified how a single transaction is represented.

Reply:
I'm hoping my approach to using `List<Object>` helps resolve this.

Reply:
Implementation of Transaction class has simplified this logic